go-controller/pkg/node/healthcheck_node.go

```diff
@@ -0,0 +1,149 @@
+package node
+
+import (
+	"os"
+	"sync"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"github.com/pkg/errors"
+
+	"k8s.io/klog/v2"
+)
+
+type openflowManager struct {
+	defaultBridge         *bridgeConfiguration
+	externalGatewayBridge *bridgeConfiguration
+	// flow cache, use map instead of array for readability when debugging
+	flowCache     map[string][]string
+	flowMutex     sync.Mutex
+	exGWFlowCache map[string][]string
+	exGWFlowMutex sync.Mutex
+	// channel to indicate we need to update flows immediately
+	flowChan chan struct{}
+}
+
+func (c *openflowManager) updateFlowCacheEntry(key string, flows []string) {
+	c.flowMutex.Lock()
+	defer c.flowMutex.Unlock()
+	c.flowCache[key] = flows
+}
+
+func (c *openflowManager) deleteFlowsByKey(key string) {
+	c.flowMutex.Lock()
+	defer c.flowMutex.Unlock()
+	delete(c.flowCache, key)
+}
+
+func (c *openflowManager) updateExBridgeFlowCacheEntry(key string, flows []string) {
+	c.exGWFlowMutex.Lock()
+	defer c.exGWFlowMutex.Unlock()
+	c.exGWFlowCache[key] = flows
+}
+
+func (c *openflowManager) requestFlowSync() {
+	select {
+	case c.flowChan <- struct{}{}:
+		klog.V(5).Infof("Gateway OpenFlow sync requested")
+	default:
+		klog.V(5).Infof("Gateway OpenFlow sync already requested")
+	}
+}
+
+func (c *openflowManager) syncFlows() {
+	// protect gwBridge config from being updated by gw.nodeIPManager
+	c.defaultBridge.Lock()
+	defer c.defaultBridge.Unlock()
+
+	c.flowMutex.Lock()
+	defer c.flowMutex.Unlock()
+
+	flows := []string{}
+	for _, entry := range c.flowCache {
+		flows = append(flows, entry...)
+	}
+
+	_, stderr, err := util.ReplaceOFFlows(c.defaultBridge.bridgeName, flows)
+	if err != nil {
+		klog.Errorf("Failed to add flows, error: %v, stderr, %s, flows: %s", err, stderr, c.flowCache)
+	}
+
+	if c.externalGatewayBridge != nil {
+		c.exGWFlowMutex.Lock()
+		defer c.exGWFlowMutex.Unlock()
+
+		flows := []string{}
+		for _, entry := range c.exGWFlowCache {
+			flows = append(flows, entry...)
+		}
+
+		_, stderr, err := util.ReplaceOFFlows(c.externalGatewayBridge.bridgeName, flows)
+		if err != nil {
+			klog.Errorf("Failed to add flows, error: %v, stderr, %s, flows: %s", err, stderr, c.exGWFlowCache)
+		}
+	}
+}
+
+// checkDefaultOpenFlow checks for the existence of default OpenFlow rules and
+// exits if the output is not as expected
+func (c *openflowManager) Run(stopChan <-chan struct{}, doneWg *sync.WaitGroup) {
```

Comment:
i'm not sure this file name makes sense. The openflow manager isnt a healthcheck, its programming flows into OVS

Reply:
well I guess part of it is a healthcheck as it makes sure ports exist...but the primary purpose is to program flows in shared gw bridge.

Reply:
@trozet OK; it used to be in healthcheck.go though; was it wrong there?

Reply:
Split the openflow manager code into its own file.